### PR TITLE
731 per user language

### LINKED ISSUE #731: Language setting :: store in server
The language is currently set only in local storage, and is therefore specific to a browser.
Ideally the language would be a property of a user or a site, and configured server side.

Would it be possible to 
- take the user language property from the mSupply user, sync this to omSupply and configure the language on login? or
- select a language prior to logging in, in the way that mobile does?
- have the user language as a default, but retain the ability for users to select another language?

### PR DESCRIPTION
**Pull request by mark-prins**
Closes #731 

Takes the language from the mSupply user on login. Uses that, if supported, as the language for the site.
Have removed the browser language detection, which was intended as a stop gap solution and is in conflict with per-user-language. Rather than allowing both and trying to determine which should override, I've taken the user setting.

Have kept the manual language selection, as I think this may be useful for demo purposes.

### CODE REVIEW

File: server/repository/migrations/postgres/2022-10-11T10-22_v102/up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TYPE key_type ADD VALUE IF NOT EXISTS 'SETTINGS_DISPLAY_CUSTOM_THEME';
```
Comment (mark-prins) [inline server/repository/migrations/postgres/2022-10-11T10-22_v102/up.sql:1]: renamed file, which incorrectly had `100` rather than `10` in the name. To avoid migration issues, have reintroduced the `NOT EXISTS` check

File: server/repository/migrations/postgres/2022-10-27T09-15_v103/up.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE user_account ADD COLUMN IF NOT EXISTS "language" varchar NOT NULL DEFAULT 'en';
```
Comment (mark-prins) [inline server/repository/migrations/postgres/2022-10-27T09-15_v103/up.sql:1]: `NOT EXISTS` is there because I was testing and ran migration after adding the column.. kept it in to be safe

File: server/repository/src/db_diesel/user_row.rs [hunk 3]
```diff
@@ -97,6 +99,7 @@ pub fn unknown_user() -> User {
             username: "unknown".to_string(),
             hashed_password: "unknown".to_string(),
             email: Some("unknown@sussol.net".to_string()),
+            language: String::from("en"),
```
Comment (mark-prins) [inline server/repository/src/db_diesel/user_row.rs:102]: there are too many `en` string literals for my liking.. but 🤷

File: server/service/src/login.rs [hunk 4]
```diff
@@ -229,6 +229,16 @@ impl LoginService {
                 "" => None,
                 _ => Some(user_info.user.e_mail.to_string()),
             },
+            language: match user_info.user.language {
```
Comment (mark-prins) [inline server/service/src/login.rs:232]: I _really_  wanted to make an enum instead but couldn't see how to make it work proper

Comment (andreievg) [reply server/service/src/login.rs:232]: Would be like this one: https://github.com/openmsupply/open-msupply/compare/731-per-user-language...731-language-enum

At this point on front end would need to do some mapping from the enum.

I like enums, although I am getting somewhat convinced that in some cases they are not ideal, but for me it's similar to abbreviations issue (or escaping rust/db keywords, rather then re-naming things to avoid conflict):
* Better not to abbreviate to reduce ambiguity (I think all agree)
* Ohh but in some cases we can abbreviate when it's obvious (or when non abbreviated is too long) <- and here is where it all breaks for me, what is obvious ?, to whom? is it going to continue being obvious ? 

So basically, in my view, it's easier if we just draw the line (less discussion, don't need to think if particular abbreviation is obvious, etc..), never abbreviate or always enums if the variants are known at compile time. Anyways that's just my opinion.

**Review by andreievg — APPROVED** [review]
lgtm, 🙏

File: server/service/src/login.rs (hunk 4 above)
Comment (mark-prins) [reply server/service/src/login.rs:232]: that's awesome! thanks - have added in e3b00dd6b and popped in the f/e mapping of Enum to strings
